python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+import uuid
+from enum import Enum
+from typing import Tuple
+
+from github import Github
+
+from .common_tool_methods import *
+from .common_validations import *
+
+MULTI_EXTENSION_SQL = "src/test/regress/sql/multi_extension.sql"
+CITUS_CONTROL = "src/backend/distributed/citus.control"
+MULTI_EXTENSION_OUT = "src/test/regress/expected/multi_extension.out"
+CONFIG_PY = "src/test/regress/upgrade/config.py"
+DISTRIBUTED_DIR_PATH = "src/backend/distributed"
+CONFIGURE_IN = "configure.in"
+CONFIGURE = "configure"
+CITUS_CONTROL_SEARCH_PATTERN = r"^default_version*"
+
+MULTI_EXT_MAJOR_SEARCH_PATTERN = r"^\s*\d{1,2}\.\d{1,2}devel$"
+MULTI_EXT_MINOR_PATCH_SEARCH_PATTERN = r"^\s*" + CITUS_VERSION_PATTERN
```

Comment:
Evaluation of `MULTI_EXT_MINOR_PATCH_SEARCH_PATTERN` seems little bit odd, since we close the regex with `$` in the variable that we use (`CITUS_VERSION_PATTERN`). Also left some new naming suggestions for them.

```suggestion
MULTI_EXT_DEVEL_SEARCH_PATTERN = r"^\s*" + CITUS_MINOR_VERSION_PATTERN + "devel$"
MULTI_EXT_PATCH_SEARCH_PATTERN = r"^\s*" + CITUS_PATCH_VERSION_PATTERN + r"$"

(in common_validations.py)
CITUS_PATCH_VERSION_PATTERN = CITUS_MINOR_VERSION_PATTERN + r".\d{1,2}"
CITUS_MINOR_VERSION_PATTERN = r"\d{1,2}\.\d{1,2}"
```

[means MULTI_EXT_MAJOR_SEARCH_PATTERN -> MULTI_EXT_DEVEL_SEARCH_PATTERN & MULTI_EXT_MINOR_PATCH_SEARCH_PATTERN -> MULTI_EXT_PATCH_SEARCH_PATTERN
(second one is my bad from the previous round)]

Reply:
Fixed